sofar/sofastream.py

```diff
@@ -0,0 +1,192 @@
+    :class:`SofaStream`-instance, e.g. ``'.GLOBAL_RoomType'``.
+    :class:`SofaStream` uses the namespace of sofar – for more information on
+    available data and its associated names refer to the :ref:`conventions
+    section <conventions_introduction>`. Also make sure to check the examples
+    below on how to use :class:`SofaStream`.
+
+    Parameters
+    ----------
+    filename : str
+        Full path to a SOFA-file
+
+    Returns
+    --------
+    sofastream : SofaStream
+
+    Notes
+    -----
+    Accessing data:
+
+    SofaStream is supposed to be used within a with-statement. To acces data
+    append the corresponding name to your instance of :class:`SofaStream`.
+
+    Depending on the type of the requested data you either get the values
+    directly or need to slice the returned variable:
+
+    - If an attribute is called, its value is returned directly.
+    - If a variable is called, a netCDF4-variable is returned. To get the
+      values it needs to be sliced (see examples).
+
+    Examples
+    --------
+    Get an attribute from a SOFA-file:
+
+        >>> from sofar import SofaStream
+        >>> filename = "path/to/file.sofa"
+        >>> with SofaStream(filename) as file:
+        >>>     data = file.GLOBAL_RoomType
+        >>>     print(data)
+        free field
+
+    Get a variable from a SOFA-file:
+
+        >>> with SofaStream(filename) as file:
+        >>>     data = file.Data_IR
+        >>>     print(data)
+        <class 'netCDF4._netCDF4.Variable'>
+        float64 Data.IR(M, R, N)
+        unlimited dimensions:
+        current shape = (11950, 2, 256)
+        filling on, default _FillValue of 9.969209968386869e+36 used
+
+    What is returned is a `netCDF-variable`. To access the values (in this
+    example the IRs) the variable needs to be sliced:
+
+        >>> with SofaStream(filename) as file:
+        >>>     data = file.Data_IR
+        >>>     # get all values
+        >>>     all_irs = data[:]
+        >>>     print(all_irs.shape)
+        (11950, 2, 256)
+        >>>     # get data from first channel
+        >>>     specific_irs = data[:,0,:]
+        >>>     print(specific_irs.shape)
+        (11950, 256)
+    """
+
+    def __init__(self, filename):
+        self._filename = filename
+
+    def __enter__(self):
+        self._file = Dataset(self._filename, mode="r")
+        return self
+
+    def __exit__(self, *args):
+        self._file.close()
+
+    def __getattr__(self, name):
+        # get netCDF4-attributes and -variable-keys from SOFA-file
+        dset_variables = np.array([key for key in self._file.variables.keys()])
+        dset_attributes = np.asarray(self._file.ncattrs())
+
+        # remove delimiter from passed sofar-attribute
+        name_netcdf = name.replace(
+            'GLOBAL_', '').replace('Data_', 'Data.')
+
+        # Handle variable-attributes (e.g. '_Units' and '_Type')
+        var_attr = None
+        if "_Units" in name_netcdf or "_Type" in name_netcdf:
```

Comment:
I think after converting to a netcdf name, any '_' will indicate you are looking at a variable attribute. Does the following work? It would be more general;
```suggestion
        if "_" in name_netcdf:
```